packages/mongodb-chatbot-ui/src/Message.tsx

```diff
@@ -130,41 +162,81 @@ export const Message = ({
   const user = useUser();
   const info = getMessageInfo(messageData, user);
 
+  const [ratingCommentStatus, setRatingCommentStatus] =
+    useState<RatingCommentStatus>("none");
+
+  const [ratingCommentErrorMessage, setRatingCommentErrorMessage] = useState<
+    string | undefined
+  >();
+
+  async function submitRatingComment(commentText: string) {
+    if (!commentText) {
+      return;
+    }
+    try {
+      await conversation.commentMessage(messageData.id, commentText);
+      setRatingCommentStatus("submitted");
+    } catch (err) {
+      setRatingCommentErrorMessage(
+        "Oops, there was an issue submitting the response to the server. Please try again."
+      );
+    }
+  }
+  function abandonRatingComment() {
+    setRatingCommentStatus("abandoned");
+  }
+
   return (
     <Fragment key={messageData.id}>
       <LGMessage
         baseFontSize={16}
         isSender={info.isSender}
-        messageRatingProps={
-          showRating
-            ? {
-                className: styles.message_rating,
-                description: "How was the response?",
-                onChange: (e) => {
-                  const value = e.target.value as MessageRatingProps["value"];
-                  if (!value) {
-                    return;
-                  }
-                  conversation.rateMessage(
-                    messageData.id,
-                    value === "liked" ? true : false
-                  );
-                },
-                value:
-                  messageData.rating === undefined
-                    ? undefined
-                    : messageData.rating
-                    ? "liked"
-                    : "disliked",
-              }
-            : undefined
-        }
+        messageRatingProps={{
+          className: styles.message_rating,
+          description: "How was the response?",
+          onChange: async (e) => {
+            const value = e.target.value as MessageRatingProps["value"];
+            if (!value) {
+              return;
+            }
+            if (ratingCommentStatus === "submitted") {
+              // Once a user has submitted a comment for their rating we don't want them to be able to change their rating
```

Comment:
why not let them update?

Reply:
At a basic level because that's how it's designed https://www.figma.com/file/L41rvEwQmJJ8VzO4hCPc7Y/LeafyGreen-Chat-Library?type=design&node-id=1-493&mode=design&t=1vkDz6xoVoPuaIIH-0

The motivation being:

1. User rates the message NEGATIVE
2. User submits a comment like "This was bad because it recommended PHP when I asked for Python"
3. User re-rates the message POSITIVE
4. Now there's a mismatch between the rating and the (uneditable) comment

Reply:
ok, my instinct would be to have it all be editable but i defer to designs